src/subscription/giftcards/RedeemGiftCardWizard.js

```diff
@@ -394,6 +392,12 @@ export function loadRedeemGiftCardWizard(giftCardInfo: GiftCardRedeemGetReturn,
 				componentClass: RedeemGiftCardPage
 			}
 		]
-		return createWizardDialog(giftCardRedeemData, wizardPages, () => Promise.resolve(m.route.set("/login"))).dialog
+		return createWizardDialog(giftCardRedeemData, wizardPages,
+			() => {
+				const route = giftCardRedeemData.credentialsMethod === "signup"
+					? `/login?loginWith=${giftCardRedeemData.mailAddress()}`
```

Comment:
I don't like string manipulation in url, here's how we do it usually:
https://github.com/tutao/tutanota/blob/7f8fe5569829be3bcce5c47d5febad9fa6f7a7f8/src/login/RecoverLoginDialog.js#L125
mithril will `encodeURIComponent()` as needed

Reply:
Thanks, much nicer!